### PR TITLE
Fix floating point exception when loading from a small events log file using the loader

### PR DESCRIPTION
**Pull request by skambha**
Issue:  Floating point exception when using the loader to load some data from a small events log file

- Fixed the bps calculation to guard against div by 0
- Tested this manually with the different size logs we have from our experiment.

If you have suggestions or a recommended way to add a unit test for the loader, I'd be glad to look into it.  Please let me know. Thanks.

TESTING:
For now, I have tested this manually with the logs we have from our experiment.

**With Fix:**
 ```
./loader --events=/data_location/experiments/mitoses/jan/events.out.tfevents.xyz.com --db=/db_location/tb/db/test14db --experiment_name=test --run_name=run1 --user_name=s
<snipped>
2018-03-29 10:58:59.092540: I tensorflow/contrib/tensorboard/db/loader.cc:116] Loaded 624,975 bytes with 31 records at 2,130,206 bps
```

Empty events file: 
```
./loader --events=test --db=/db_location/tb/db/test14db --experiment_name=test --run_name=run1 --user_name=s
<snipped>
2018-03-29 12:20:03.849826: I tensorflow/contrib/tensorboard/db/loader.cc:116] Loaded 0 bytes with 0 records at 0 bps
```

**Without Fix:**
```
./loader --events=/data_location/experiments/mitoses/jan/events.out.tfevents.xyz.com --db=/db_location/tb/db/test14db --experiment_name=test --run_name=run1 --user_name=s
2018-03-13 11:53:35.910348: I tensorflow/contrib/tensorboard/db/loader.cc:75] Opening SQLite file: /db_location/tb/db/test14db
2018-03-13 11:53:35.911059: I tensorflow/contrib/tensorboard/db/loader.cc:82] Initializing TensorBoard schema
2018-03-13 11:53:35.914771: I tensorflow/contrib/tensorboard/db/loader.cc:85] Creating SummaryDbWriter
2018-03-13 11:53:35.914792: I tensorflow/contrib/tensorboard/db/loader.cc:91] Loading TF event log: /data_location/experiments/mitoses/jan/events.out.tfevents.xyz.com
Floating point exception: 8
```

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

File: tensorflow/contrib/tensorboard/db/loader.cc
Comment (jart) [inline tensorflow/contrib/tensorboard/db/loader.cc]: This creates an additional log line. Maybe we could have a ternary instead like `"bps=" << (elapsed == 0 ? 0 : (uint64)(offset / (elapsed / 1000000.0)))`.

**Review by jart — CHANGES_REQUESTED** [review]
Good catch! Thank you for participating in the development of this functionality!

File: tensorflow/contrib/tensorboard/db/loader.cc
Comment (skambha) [reply tensorflow/contrib/tensorboard/db/loader.cc]: Thank you for the review. I moved the bps to one log line like before and used the ternary as you suggest.

I added a condition to report the bps only if elapsed is greater than 0  because it did not seem right to say bps of 0 if 'n' bytes were read and elapsed was 0. 

How about I do this: 
(elapsed == 0 ? offset : static_cast<uint64>(offset / (elapsed / 1000000.0)))

So if elapsed is 0, we will report bps as the bytes read (ie offset) as that many bytes were processed in a sec. 

————
The log line will look like as before: 
2018-03-28 21:02:17.282332: I tensorflow/contrib/tensorboard/db/loader.cc:114] Loaded 590,308 bytes with 29 records at 1907172 bps

**Comment by skambha** [comment]
Thank you for the review.  I have updated the code to use the commas and the log line like before. The bps calculation details are as listed in my earlier comment.  I used a variable to do the bps computation instead as it looked a bit more readable than putting the ternary code in the log line itself.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @jart: It has been 14 days with no activity and the `awaiting review` label was assigned. Can you please take a look?

**Review by jart — APPROVED** [review]

**Comment by skambha** [comment]
There is a timeout in the CI test and I think it is not related to this change. 
Is it possible to get these tests to run again?   Thanks.

**Comment by jart** [comment]
Assigning to @protoget who's on sync duty.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 22 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 37 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by skambha** [comment]
@protoget , The changes are reviewed and approved but it is waiting for CI tests to be rerun. Is it possible to get these tests to run again.  If there is anything else that is needed before this can get merged, please let me know. Thanks for your help.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
I'm seeing a test failure on MacOS that may be related-- 

```
tensorflow/contrib/lite/profiling/profiler_test.cc:34
The difference between expected_ms and duration_ms is 10.062999999999988, which exceeds eps_ms, where
expected_ms evaluates to 500,
duration_ms evaluates to 510.06299999999999, and
eps_ms evaluates to 10.
```
@skambha would that be expected from your changes? @shashishekhar , can you comment as the author of that test?

**Comment by skambha** [comment]
@karmel,  Thanks for triggering a test run and looking into this. 
I think that failure is not related to this pr's changes.  Also, fwiw, I ran the test on my macOS and it passed.

**Comment by karmel** [comment]
Thanks, @skambha -- rerunning now.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Pull request merged**
